libs/portals/admin/ids-admin/src/components/Client/Permissions.tsx

```diff
@@ -1,86 +1,78 @@
+import React, { useContext, useState } from 'react'
 import { ClientFormTypes } from '../forms/EditApplication/EditApplication.action'
 import { ShadowBox } from '../ShadowBox/ShadowBox'
-
-type Permission = {
-  id: string
-  label: string
-  description: string
-  api: string
-  locked?: boolean
-}
+import { useParams } from 'react-router-dom'
+import { ClientContext } from '../../shared/context/ClientContext'
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import AddPermissions from '../forms/AddPermissions/AddPermissions'
+import { getTranslatedValue } from '@island.is/portals/core'
 
 interface PermissionsProps {
-  data?: Permission[]
+  allowedScopes?: AuthAdminClientAllowedScope[] | undefined
 }
 
-const mockData: NonNullable<PermissionsProps['data']> = [
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance:overview',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin:full',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Skattskýrslur',
-    id: '@skatturinn.is/skattskyrslur',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Skatturinn',
-    locked: true,
-  },
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-]
-
-function Permissions({ data = mockData }: PermissionsProps) {
-  const { formatMessage } = useLocale()
+function Permissions({ allowedScopes }: PermissionsProps) {
+  const { formatMessage, locale } = useLocale()
   const [isModalVisible, setIsModalVisible] = useState(false)
+  const params = useParams()
+  const [permissions, setPermissions] = useState<AuthAdminClientAllowedScope[]>(
+    allowedScopes ?? [],
+  )
+  const {
+    addedScopes,
+    removedScopes,
+    setRemovedScopes,
+    setAddedScopes,
+  } = useContext(ClientContext)
+
+  const tenant = params['tenant']
+
+  const handleModalOpen = () => {
+    setIsModalVisible(true)
+  }
 
   const handleModalClose = () => {
     setIsModalVisible(false)
   }
 
-  const handleModalOpen = () => {
-    setIsModalVisible(true)
+  // If removing permissions, we want to remove it from both the permissions and addedPermissions array and add it to the removedPermissions array if it doesn't exist there
+  const handleRemovedPermissions = (
```

Comment:
~The function is handling removal of a single permission
```suggestion
  const handleRemovedPermission = (
```